### ISSUE TITLE
1.8.0-beta: ts.transpile now errors for .js files

### ISSUE THREAD

**Issue by frankwallis** [Microsoft/TypeScript#6766]
The ```ts.transpile``` method now throws an error when given a file with a ```.js``` extension:

test.js:
```
var ts = require("typescript");
var diags = [];
var output = ts.transpile("const a = 10; export default a;", { module: 1 }, "input.js", diags);
console.log(output);
console.log(JSON.stringify(diags));
```

output:
```
Heracles:plugin-typescript frank$ node test.js
/Users/frank/work/plugin-typescript/node_modules/typescript/lib/typescript.js:1682
                throw new Error("Debug Failure. False expression: " + (message || "") + verboseDebugString);
                ^

Error: Debug Failure. False expression: Output generation failed
    at Object.assert (/Users/frank/work/plugin-typescript/node_modules/typescript/lib/typescript.js:1682:23)
    at transpileModule (/Users/frank/work/plugin-typescript/node_modules/typescript/lib/typescript.js:47824:18)
    at Object.transpile (/Users/frank/work/plugin-typescript/node_modules/typescript/lib/typescript.js:47832:22)
    at Object.<anonymous> (/Users/frank/work/plugin-typescript/test.js:3:4)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Function.Module.runMain (module.js:457:10)
    at startup (node.js:138:18)
```
The actual error causing this is TS5055:
```
Cannot write file 'input.js' because it would overwrite input file.
```
This breaks e.g. the systemjs built-in typescript transpilation:(https://github.com/systemjs/systemjs/blob/6243d0c7700ca802699796cfae0f9e213072e077/dist/system.src.js#L1210)
Looks to be caused by the changes made for ```allowJs``` option in https://github.com/Microsoft/TypeScript/pull/5471.

**Comment by denis-sokolov** [comment]
May we see a 1.8 branch with this fix? Currently the entire 1.8 branch is broken for us, a 1.8.3 release would be appreciated.

**Comment by mhegazy** [comment]
@denis-sokolov this has been ported in release-1.8 a while back in https://github.com/Microsoft/TypeScript/pull/6791.

do you see the same issue on `typescript@1.8.2`?

**Comment by denis-sokolov** [comment]
@mhegazy, ah, sorry, I did not find that it was ported. Yes, I did all my testing on 1.8.2, I never had 1.8.0 installed. I shall double-check, just in case. Anything else to be on the lookout for?

**Comment by mhegazy** [comment]
We believe this issue has been fixed. if you are still running into issues please file a new issue and give us more information to be able to diagnose it.

**Comment by denis-sokolov** [comment]
For other people who stumble here. My problem was caused by the [es6-default-import-commonjs](https://github.com/Microsoft/TypeScript/issues/5565) problem, and was solved by upgrading `ts-loader` from [0.7.2 to 0.8.1](https://github.com/TypeStrong/ts-loader/blob/master/CHANGELOG.md).
